test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_mnist_inductor.py

```diff
@@ -137,6 +137,9 @@ def _test_mnist_distributed(
         framework_version=framework_version,
         image_uri=ecr_image,
         sagemaker_session=sagemaker_session,
+        env={
```

Comment:
Let's move this env setting to version specific to PyTorch 2.4 so we don't break functionality on older PyTorch version tests.

We can create a new helper function with something like this
```
def set_authorization_token(ecr_image):
    _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
    return {"TOKEN": "false"} if Version(image_framework_version) in SpecifierSet(">=2.4") else None
```

And then call it within the pytest functions (such as `test_mnist_distributed_cpu_inductor`) with
```
    env = set_authorization_token(ecr_image)
    function_args = {
        "framework_version": framework_version,
        "instance_type": instance_type,
        "model_dir": model_dir,
        "mnist_script": mnist_cpu_script,
        "env": env
    }
```